internal/index_read.go

```diff
@@ -165,15 +166,22 @@ func readIntervals(r io.Reader, typ string) ([]bgzf.Offset, error) {
 	if n == 0 {
 		return nil, nil
 	}
-	var vOff uint64
 	offsets := make([]bgzf.Offset, n)
-	for i := range offsets {
-		err := binary.Read(r, binary.LittleEndian, &vOff)
+	// chunkSize determines the number of offsets consumed by each binary.Read.
+	const chunkSize = 512
+	var vOffs [chunkSize]uint64
+	for i := 0; i < int(n); i += chunkSize {
+		// Limit the size of vOffs on final iteration.
+		l := min(int(n)-i, chunkSize)
+		err := binary.Read(r, binary.LittleEndian, vOffs[:l])
```

Comment:
I made those changes except for removing `l` because that's used in the following loop.